### PR TITLE
build:Modify GitHub releases URL to aws s3

### PR DESCRIPTION
**Pull request by Shuimo03**
Signed-off-by: Shuimo03 <wu1998102@gmail.com>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description

Because the GitHub releases URL was slow to download before, switching to AWS S3 will change the download speed.

before:

now

## Related Issues

issuees:https://github.com/devstream-io/devstream/issues/1314

### CODE REVIEW

File: internal/pkg/upgrade/upgrade.go [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package upgrade
 
 import (
+	"github.com/devstream-io/devstream/pkg/util/downloader"
```
Comment (daniel-hutao) [inline internal/pkg/upgrade/upgrade.go:4]: Please use `make fmt` before commit.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

File: internal/pkg/upgrade/upgrade.go (hunk 1 above)
Comment (Shuimo03) [reply internal/pkg/upgrade/upgrade.go:4]: ok

**Comment by daniel-hutao** [comment]
DevStream community follows the [conventional commits specification](https://www.conventionalcommits.org/eXn/v1.0.0/). Please read our [doc](https://docs.devstream.io/en/latest/development/commit-messages/) and update your commit message.

File: internal/pkg/upgrade/upgrade.go [hunk 2]
```diff
@@ -62,21 +64,22 @@ func Upgrade(continueDirectly bool) error {
 		return err
 	}
 
-	ltstReleaseTagName, err := ghClient.GetLatestReleaseTagName()
+	//ltstReleaseTagName, err := ghClient.GetLatestReleaseTagName()
+	lastReleaseTagName, err := ghClient.GetLatestReleaseTagName()
```
Comment (aFlyBird0) [inline internal/pkg/upgrade/upgrade.go:68]: `ltst` is short for `latest` so there are no misspellings here.

But there is really something that needs to be changed here, and we should get the latest version directly through s3, not github.

You can `GET` this URL to get the latest version from s3: `https://download.devstream.io/latest_version`

**Review by aFlyBird0 — CHANGES_REQUESTED** [review]
Thank you for your contribution! @Shuimo03 

I've left some comments in the hope that we can continue to improve the code together.

One more tiny one that has little impact on the code itself.

It looks like this branch of yours hasn't pulled the latest code from the `main` branch of dtm in a long time. I recommend running `git fetch upstream && git rebase upstream/main`.

For more guidance on GitHub collaboration you can read this blog: https://blog.devstream.io/posts/open-a-pr-in-github/

**Comment by aFlyBird0** [comment]
I noticed that `lint commit message` CI failed, this article could help you:

DevStream community follows the [conventional commits specification](https://www.conventionalcommits.org/en/v1.0.0/). Please read our [doc](https://docs.devstream.io/en/latest/development/git-workflow/commit-messages/) and update your commit message.

File: internal/pkg/upgrade/upgrade.go (hunk 2 above)
Comment (Shuimo03) [reply internal/pkg/upgrade/upgrade.go:68]: I will pay attention to it, I will look at it later😀

**Comment by aFlyBird0** [comment]
After changing your code and updating commit messages, you could use `git rebase` to make commit history more clear:

https://blog.devstream.io/posts/open-a-pr-in-github/#622-ide-%E9%87%8C%E5%90%88%E5%B9%B6-commits

And, at last, just run `git push -f`

File: internal/pkg/upgrade/upgrade.go (hunk 2 above)
Comment (aFlyBird0) [reply internal/pkg/upgrade/upgrade.go:68]: > > `ltst` is short for `latest` so there are no misspellings here.
> > But there is really something that needs to be changed here, and we should get the latest version directly through s3, not github.
> > You can `GET` this URL to get the latest version from s3: `https://download.devstream.io/latest_version`
> 
> @aFlyBird0 I think the `ltst` is a bad abbreviation. How about leave `latest` here?

lgtm

**Review by aFlyBird0 — APPROVED** [review]

**Review by daniel-hutao — APPROVED** [review]
lgtm

**Pull request merged**